### DISCUSSION TITLE
Add right click to copy selection and paste from clipboard

### DISCUSSION THREAD

**Discussion by naiyerasif** [ghostty-org/ghostty#3438] [Feature Requests, Ideas] [answered]
This behavior is something that I've used over years (coming from [PuTTY](https://www.putty.org/)). This is how it works:

- If I select some text and right click, the text gets copied to clipboard.
- If I right click on a prompt that accepts input, the last copied text gets pasted.

Windows Terminal and iTerm2 which I use at work and home support this. It would be very handy if I can configure Ghostty to behave similarly.

**Comment by leetrout** [comment]
Related to #265 & #266

**Comment by zzxwill** [comment]
And coming from iTerm.

**Comment by 00-kat** [comment, marked as answer]
This was here first, but someone later opened the same for right click to copy and it's now been promoted to #4404.  And right click to paste would covered by #3848.

**Comment by celevra** [comment]
why ist this thread marked answered?
There is no solution to paste on right clickt, isn't it?

can't use ghostty on production without this feature

**Reply by 00-kat** [reply]
> why ist this thread marked answered?

Because this thread's request is covered by another, as the comment marked as the answer clearly states.

>  There is no solution to paste on right clickt, isn't it?

No, not yet, there is some discussion in #4404 and there is also a feature request at #3848 which covers it.
